sql/telemetry_derived/clients_daily_v6/query.sql

```diff
@@ -369,7 +369,13 @@ SELECT
   udf.mode_last(ARRAY_AGG(windows_ubr ORDER BY `timestamp`)) AS windows_ubr,
   udf.mode_last(ARRAY_AGG(fxa_configured ORDER BY `timestamp`)) AS fxa_configured,
   SUM(scalar_parent_contentblocking_trackers_blocked_count) AS trackers_blocked_sum,
-  TIMESTAMP_MICROS(DIV(MIN(`timestamp`), 1000)) AS submission_timestamp_min
+  TIMESTAMP_MICROS(DIV(MIN(`timestamp`), 1000)) AS submission_timestamp_min,
+  udf.sum_array_struct(
+    ARRAY_CONCAT_AGG(scalar_parent_browser_search_ad_clicks ORDER BY `timestamp`)
+  ) AS ad_clicks_count_all,
+  udf.sum_array_struct(
+    ARRAY_CONCAT_AGG(scalar_parent_browser_search_with_ads ORDER BY `timestamp`)
+  ) AS search_with_ads_count_all
```

Comment:
This aggregation will be more efficient without the UDF:
```suggestion
  SUM((SELECT SUM(value) FROM UNNEST(scalar_parent_browser_search_ad_clicks))) AS ad_clicks_count_all,
  SUM((SELECT SUM(value) FROM UNNEST(scalar_parent_browser_search_with_ads))) AS search_with_ads_count_all,
```

Reply:
Hi Daniel, thanks for your suggestion. I have changed my code, and it passed most of test. However there is an error in build test saying the 'scalar_parent_browser_search_ad_clicks' which is a record in `moz-fx-data-shared-prod.telemetry.main_summary_v4` doesn't exist. But I run this code successfully on Bigquery, and I checked that this code exists in the dataset. Can you help me figure out what happen here? Thanks!

Reply:
you need these changes:
```diff
diff --git a/tests/telemetry_derived/clients_daily_v6/test_aggregation/expect.yaml b/tests/telemetry_derived/clients_daily_v6/test_aggregation/expect.yaml
index 6040abb..9ab6fca 100644
--- a/tests/telemetry_derived/clients_daily_v6/test_aggregation/expect.yaml
+++ b/tests/telemetry_derived/clients_daily_v6/test_aggregation/expect.yaml
@@ -137,8 +137,11 @@
   search_count_all: 0
   search_count_contextmenu: 0
   search_count_newtab: 0
+  search_count_organic: 0
   search_count_searchbar: 0
   search_count_system: 0
+  search_count_tagged_follow_on: 0
+  search_count_tagged_sap: 0
   search_count_urlbar: 0
   session_restored_mean: 2
   sessions_started_on_this_day: 0
diff --git a/tests/telemetry_derived/clients_daily_v6/test_aggregation/main_summary_v4.schema.json b/tests/telemetry_derived/clients_daily_v6/test_aggregation/main_summary_v4.schema.json
index 1c9687d..e1a0f05 100644
--- a/tests/telemetry_derived/clients_daily_v6/test_aggregation/main_summary_v4.schema.json
+++ b/tests/telemetry_derived/clients_daily_v6/test_aggregation/main_summary_v4.schema.json
@@ -1003,5 +1003,39 @@
     "type": "STRING",
     "name": "isp_organization",
     "mode": "NULLABLE"
+  },
+  {
+    "fields": [
+      {
+        "type": "STRING",
+        "name": "key",
+        "mode": "NULLABLE"
+      },
+      {
+        "type": "INTEGER",
+        "name": "value",
+        "mode": "NULLABLE"
+      }
+    ],
+    "type": "RECORD",
+    "name": "scalar_parent_browser_search_ad_clicks",
+    "mode": "REPEATED"
+  },
+  {
+    "fields": [
+      {
+        "type": "STRING",
+        "name": "key",
+        "mode": "NULLABLE"
+      },
+      {
+        "type": "INTEGER",
+        "name": "value",
+        "mode": "NULLABLE"
+      }
+    ],
+    "type": "RECORD",
+    "name": "scalar_parent_browser_search_with_ads",
+    "mode": "REPEATED"
   }
 ]
```

Reply:
explanation: tests don't have access to production tables, so a partial schema for main_summary_v4 is provided for the test, and that schema the one missing the fields. The expected test results also change, because of the three new columns added to clients_daily_v6.